ruma-client-api/src/r0/device/delete_devices.rs

```diff
@@ -3,7 +3,9 @@
 use ruma_api::ruma_api;
 use ruma_identifiers::DeviceIdBox;
 
-use crate::r0::uiaa::{AuthData, IncomingAuthData, UiaaResponse};
+#[cfg(any(feature = "client", feature = "server"))]
+use crate::r0::uiaa::UiaaResponse;
```

Comment:
I don't think we should import error types conditionally. If you want to have these warnings fixed, you could emit `type _SilenceUnusedError = #error_ty;` from `ruma_api!`.

Reply:
One other solution I have been thinking about is to force the use of at least one of `client` or `server`. Unless it makes sense to use neither?